Tribler/Core/Libtorrent/LibtorrentDownloadImpl.py

```diff
@@ -287,6 +311,13 @@ def network_create_engine_wrapper(self, pstate, initialdlstatus=None):
             if self.handle:
                 self.set_selected_files()
 
+                # set_selected_files sets priorities to 1, so we must set
+                # share_mode again, but first we must unset it, otherwise
+                # set_share_mode doesn't do anything
```

Comment:
I think this is an ugly workaround. I would rather refactor `set_selected_files` (i.e. add a boolean parameter whether the priorities should be changed or not).